### ISSUE TITLE
False Positive: 	WordPress.DB.PreparedSQL.NotPrepared

### ISSUE THREAD

**Issue by nikelone** [10up/plugin-check#906]
Thanks a lot for this plugin. This helps really.

I think I discovered a bug though:
For this code:
```PHP    
$query = "SELECT 
            second_level_domain,
            MIN(intercepted_at) AS first_detected,
            MAX(intercepted_at) AS last_detected,
            SUM(CASE WHEN intercepted_at >= DATE_SUB(NOW(), INTERVAL 1 HOUR) THEN 1 ELSE 0 END) AS count_1_hour,
            SUM(CASE WHEN intercepted_at >= DATE_SUB(NOW(), INTERVAL 24 HOUR) THEN 1 ELSE 0 END) AS count_24_hours,
            SUM(CASE WHEN intercepted_at >= DATE_SUB(NOW(), INTERVAL 7 DAY) THEN 1 ELSE 0 END) AS count_7_days
         FROM $table_name
         GROUP BY second_level_domain
         ORDER BY $orderby $order
         LIMIT %d, %d
    ";

        $summary = $wpdb->get_results($wpdb->prepare($query, $offset, $per_page));
```

I get this error: 

197 | 54 | ERROR | WordPress.DB.PreparedSQL.NotPrepared | Use placeholders and $wpdb->prepare(); found $query

Where this line: $summary = $wpdb->get_results($wpdb->prepare($query, $offset, $per_page)); is in 197 and $query starts at column 54. 

![Image](https://github.com/user-attachments/assets/9f102f75-928e-4bd3-82f4-ff1f8bc46d56)

**Comment by swissspidy** [comment]
Hi there,

This sniff is coming from https://github.com/WordPress/WordPress-Coding-Standards, so if you believe to have found a bug in it, you should report it there.

That said, it looks like you're not really preparing the query correctly. `%able_name`, `$orderby`, `$order` appear to be arbitrary unsanitized variables. Simply wrapping` prepare()` around such a query does not make it secure.

I recommend consulting documentation on database query escaping. And if you really found a bug in the sniff, you know where to report it.

**Comment by nikelone** [comment]
Hi,
yes. thanks a lot for your quick reply and your instructions.
I am a bit confused about the error message. I would have expected WordPress.DB.PreparedSQL.InterpolatedNotPrepared. But will have a deeper look and see in the linked repo.

Please excuse the wrong post here.

Kind Regards
